qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -40,10 +44,98 @@ def __init__(self, experiments, qubits, experiment_type=None):
         self._num_experiments = len(experiments)
         super().__init__(qubits, experiment_type=experiment_type)
 
+    def run_transpile(self, backend: Backend, **options) -> List[QuantumCircuit]:
+        """Run transpile and returns transpiled circuits.
+
+        Args:
+            backend: Target backend.
+            options: User provided runtime options.
+
+        Returns:
+            Transpiled circuit to execute.
+
+        Note:
+            This is transpile method for the composite experiment subclass.
+            This internally calls the transpile method of the nested experiments and
+            flattens the list of sub circuits generated by each experiment.
+            Note that transpile is called for individual circuit, and thus transpile
+            configurations and hook methods are separately applied.
+
+            No transpile configuration assumed for composite experiment object itself.
+        """
+        # Transpile each sub experiment circuit
+        circuits = list(map(lambda expr: expr.run_transpile(backend), self._experiments))
+
+        # This is not identical to the `num_qubits` when the backend is AerSimulator.
+        # In this case, usually a circuit qubit number is determined by the maximum qubit index.
+        n_qubits = max(max(sub_circ.num_qubits for sub_circ in sub_circs) for sub_circs in circuits)
+
+        # merge circuits
+        return self._flatten_circuits(circuits, n_qubits)
+
+    def run_analysis(
+        self, experiment_data: ExperimentData, job: BaseJob = None, **options
+    ) -> ExperimentData:
+        """Run analysis and update ExperimentData with analysis result.
+
+        Args:
+            experiment_data: The experiment data to analyze.
+            job: The future object of experiment result which is currently running on the backend.
+            options: Additional analysis options. Any values set here will
+                override the value from :meth:`analysis_options` for the current run.
+
+        Returns:
+            An experiment data object containing the analysis results and figures.
+
+        Raises:
+            QiskitError: When the experiment data format is not for the composite experiment.
+
+        Note:
+            This is analysis method for the composite experiment subclass.
+            This internally calls the analysis method of the nested experiments and
+            outputs a representative data entry for the composite analysis.
+            Note that analysis is called for individual experiment data, and thus analysis
+            configurations and hook methods are separately applied.
+
+            No analysis configuration assumed for composite experiment object itself.
+        """
+        if not isinstance(experiment_data, CompositeExperimentData):
```

Comment:
Replace `CompositeExperimentData` by `self.__experiment_data__`, to cover the case where the user wrote her in-house experiment data class

Reply:
Thanks. Done in https://github.com/Qiskit/qiskit-experiments/pull/380/commits/2abc6d87ba38e22247a998af40defa4bd7bbc716